src/js/engagement_view/src/components/graphDisplay/GraphDisplay.tsx

```diff
@@ -249,6 +241,10 @@ const GraphDisplay = ({ lensName, setCurNode }: GraphDisplayProps) => {
         const LABEL_NODE_MARGIN = 12;
         const start = link.source;
         const end = link.target;
+
+        link.fx = link.x;
+        link.fy = link.y;
+
```

Comment:
# try this change after you do everything else
and i'm not opposed if you do it in a followup PR, as long as you do _do_ it 
---

We can actually get correct types on `constlinkStyling` fwiw
https://github.com/vasturiano/react-force-graph/blob/master/src/packages/react-force-graph-2d/index.d.ts#L71
```
it's a CanvasCustomRenderFn<LinkObject>;
which boils down to a 
(link: LinkObject, canvasContext: CanvasRenderingContext2D) => {
```

you'll have to import LinkObject type from react-force-graph-2d
and you'll have to import CanvasRenderingContext2D from some presupplied web api typescript gives you
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D

---
that said, here I see you doing like `link.x, link.y` whereas LinkObject in the linked source code says it only has properties `.source` and `.target`, so, I'm a tad confused how that all works...

Reply:
This is dead code that shouldn't be in here, sorry - was testing something and forgot to remove this.You're right with the  source and target.